src/Forms/__tests__/Checkbox.test.tsx

```diff
@@ -0,0 +1,342 @@
+import {
+  SinonSpy,
+  spy,
+} from 'sinon';
+import {
+  BoundFunction,
+  GetByText,
+  QueryByText,
+  render,
+  fireEvent,
+} from 'test-utils';
+import React, { useRef } from 'react';
+import Checkbox from 'Forms/Checkbox';
+import { strictEqual } from 'assert';
+import userEvent from '@testing-library/user-event';
+import { POSITION } from 'Forms/Label';
+import { Button } from 'Buttons';
+import { VARIANT } from 'Theme';
+
+describe('Checkbox', function () {
+  let getByText: BoundFunction<GetByText>;
+  let queryByText: BoundFunction<QueryByText>;
+  let changeSpy: SinonSpy;
+  context('when errorMessage prop is present', function () {
+    const checkboxId = 'testCheckbox';
+    const errorMessage = 'Please select this checkbox before proceeding.';
+    beforeEach(function () {
+      changeSpy = spy();
+      ({ getByText } = render(
+        <Checkbox
+          id={checkboxId}
+          checked={false}
+          label="I agree to the terms and conditions."
+          onChange={changeSpy}
+          errorMessage={errorMessage}
+          isRequired
+        />
+      ));
+    });
+    it('renders', function () {
+      const checkboxElement = document
+        .getElementById(checkboxId) as HTMLInputElement;
+      strictEqual(!!checkboxElement, true);
+    });
+    it('calls the change handler when clicked', function () {
+      fireEvent.click(document.getElementById(checkboxId));
+      strictEqual(changeSpy.callCount, 1);
+    });
+    it('renders the correct checked value', function () {
+      const checkboxElement = document
+        .getElementById(checkboxId) as HTMLInputElement;
+      strictEqual(checkboxElement.checked, false);
+    });
```

Comment:
I'm not sure that these 3 tests need to go under the conditional like this. Unless that behavior is affected by the error message

Reply:
I reorganized all of the tests so that there was a describe block for each prop. (commit 78434c9)

Reply:
Yep. Looks great!